Modify Enh Healing Surge spell name

Nighteyez07 opened the pull request:
Enh Healing Surge is a different spell id then Resto Shaman but the same spell name. This caused a conflict within SPELLS_SHAMAN.js. Modified the Enh version to keep the peace.

MartijnHols commented:
Please make a separate PR for the Vial of Ceaseless Toxins trinket.

MartijnHols commented:
Wrt this PR: do you never use `HEALING_SURGE` in the Enhancement module? Not even in `AlwaysBeCasting`? Those should refer to the `HEALING_SURGE_ENH` spell.

I would much prefer if the Restoration `HEALING_SURGE` was renamed to `HEALING_SURGE_RESTORATION` so that there can be no confusion in the future.
I would also prefer `HEALING_SURGE_ENH` to be called `HEALING_SURGE_ENHANCE` for consistency sake.

Nighteyez07 commented:
Removed Vial trinket logic for separate PR. Updated the Healing Surge fix to reflect spec specific version and updated all the Resto references accordingly. Next Enhance update will include our Healing Surge in CPM section.
